lib/semantics/assignment.cc

```diff
@@ -363,18 +374,11 @@ void AssignmentContext::Analyze(const parser::AssignmentStmt &stmt) {
 
   // C1596 checks for polymorphic deallocation in a PURE subprogram
   // due to automatic reallocation on assignment
-  const auto &lhs{std::get<parser::Variable>(stmt.t)};
-  const auto &rhs{std::get<parser::Expr>(stmt.t)};
-  if (auto lhsExpr{AnalyzeExpr(context_, lhs)}) {
+  if (lhsExpr) {
     if (auto type{evaluate::DynamicType::From(*lhsExpr)}) {
-      if (type->IsPolymorphic() && lhsExpr->Rank() > 0) {
-        if (const Symbol * last{evaluate::GetLastSymbol(*lhsExpr)}) {
-          if (IsAllocatable(*last) && FindPureProcedureContaining(rhs.source)) {
-            evaluate::SayWithDeclaration(context_.messages(), last, at_,
-                "Deallocation of polymorphic object '%s' is not permitted in a PURE subprogram"_err_en_US,
-                last->name());
-          }
-        }
+      if (type->IsPolymorphic() && FindPureProcedureContaining(rhs.source)) {
+        Say(at_,
+            "Deallocation of polymorphic object is not permitted in a PURE subprogram"_err_en_US);
       }
       if (type->category() == TypeCategory::Derived &&
           !type->IsUnlimitedPolymorphic() /* TODO */ &&
```

Comment:
In the code that checks for deallocation of components, I don't see anything that checks to see if the components are coarrays.  Section 10.2.1.3, paragraph 13 says that only noncoarray components will be deallocated.

Reply:
Copy that; thanks.